emark/utils.py

```diff
@@ -116,6 +118,71 @@ def handle_data(self, data: str) -> None:
         if self.root:
             self.root.children.append(data)
 
+    @classmethod
+    def from_html(cls, html: str) -> HTML2TextParser:
+        """Parse HTML and return a HTML2TextParser instance.
+        >>> HTML2TextParser.from_html(html).text
+        """
+        parser = cls()
+        parser.feed(html)
+        return parser
+
+    @classmethod
+    def to_text(cls, html: str) -> str:
+        """Parse HTML and return plain text. Similar to Gmail.
+        >>> HTML2TextParser.to_text(html)
+        """
+        return str(cls.from_html(html))
+
+    @classmethod
+    def from_file(cls, path: str) -> HTML2TextParser:
+        """Parse HTML from a file and return a HTML2TextParser instance.
+        >>> HTML2TextParser.from_file(path).text
+        """
+        with open(path, "r") as file:
+            return cls.from_html(file.read())
+
+    @classmethod
+    def from_url(cls, url: str) -> HTML2TextParser:
+        """Parse HTML from a URL and return a HTML2TextParser instance. Requires requests.
+        >>> HTML2TextParser.from_url(url).text
+        """
+        import requests
+
+        return cls.from_html(requests.get(url).text)
+
+    @classmethod
+    def from_email(cls, email: str) -> HTML2TextParser:
+        """Parse HTML from an email and return a HTML2TextParser instance. Requires email.
+        >>> HTML2TextParser.from_email(email).text
+        """
+        import email
+
+        message = email.message_from_string(email)
+        if message.is_multipart():
+            for part in message.walk():
+                if part.get_content_type() == "text/html":
+                    return cls.from_html(part.get_payload())
+        else:
+            return cls.from_html(message.get_payload())
```

Comment:
This can fail, if the content isn't HTML but plain text. Maybe it's good to emit a `UserWarning` here.